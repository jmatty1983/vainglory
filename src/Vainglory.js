require('babel-polyfill');

import Api from './api';
import Http from './Http';

class Vainglory {
  constructor(apiKey = null) {
    if (!apiKey) {
      throw new Error('Missing API Key.');
    }

    const api = new Api(new Http(apiKey));
    // Exposed methods
    this.meta = api.meta;
    this.matches = api.matches;
  }

}

module.exports = Vainglory;
