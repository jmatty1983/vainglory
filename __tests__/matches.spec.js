import Vainglory from '../src/Vainglory';

const validMethods = ['collection'];

describe('Vainglory.matches', () => {
  const vainglory = new Vainglory('aaa.bbb.ccc');

  it('should contain valid methods', () => {
    validMethods.forEach((method) => {
      expect(vainglory.matches[method]).toBeDefined();
    })
  });

  // TODO
  it('.collection should get a collection of matches', async () => {
    const matches = await vainglory.matches.collection();
    expect(matches).toMatchSnapshot();
  });


});
