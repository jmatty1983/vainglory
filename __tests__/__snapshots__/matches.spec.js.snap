exports[`Vainglory.matches .collection should get a collection of matches 1`] = `"https://api.dc01.gamelockerapp.com/shards/na/matches?offset=0&limit=50&sort=createdAt&started=3hrs%20ago&ended=Now"`;

exports[`Vainglory.matches .single should get a single match 1`] = `"https://api.dc01.gamelockerapp.com/shards/na/matches?offset=0&limit=50&sort=createdAt&started=3hrs%20ago&ended=Nowmatches/0123b560-d74c-11e6-b845-0671096b3e30"`;
